### PR TITLE
Deprecate replace with categorical columns

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Matches pandas 2.2 behavior: https://github.com/pandas-dev/pandas/pull/56385

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_replace.py [hunk 1]
```diff
@@ -236,11 +250,26 @@ def test_dataframe_replace(df, to_replace, value):
...
 
-    if pd_value is None:
-        expected = pdf.replace(to_replace=pd_to_replace)
-    else:
-        expected = pdf.replace(to_replace=pd_to_replace, value=pd_value)
-    actual = gdf.replace(to_replace=gd_to_replace, value=gd_value)
+    with expect_warning_if(
+        PANDAS_GE_220
+        and isinstance(df["a"].dtype, cudf.CategoricalDtype)
+        and isinstance(to_replace, str)
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_replace.py:256]: This check is redundant with the equality comparison below.

File: python/cudf/cudf/tests/test_replace.py [hunk 2]
```diff
@@ -236,11 +250,26 @@ def test_dataframe_replace(df, to_replace, value):
...
+        and to_replace == "two"
+        and isinstance(value, str)
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_replace.py:258]: Same as above.

File: python/cudf/cudf/tests/test_replace.py (hunk 1 above)
Comment (isVoid) [reply python/cudf/cudf/tests/test_replace.py:256]: There are similar checks in other test cases too.

Comment (mroeschke) [reply python/cudf/cudf/tests/test_replace.py:256]: This was mean to protect against comparing a sting to a `Series` which results in a `Series`, and `bool(Series)` here would raise an error here. (same as below)

**Review by isVoid — APPROVED** [review]

**Pull request merged**